### PR TITLE
feat: PGP sign the checksum file in releases

### PR DESCRIPTION
**Pull request by deansheather**
Adds `coder_${version}_checksums.txt.asc` to releases. Signed with the release key, for which the public key can be found in `release.key`.

Kyle made the key and added it to GitHub Actions secrets. He will be adding it to 1password and sharing it with Ammar so we don't lose it and it stays secure.

image

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -214,6 +214,7 @@ jobs:
             ./build/*.rpm
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CODER_GPG_RELEASE_KEY_BASE64: ${{ secrets.CODER_GPG_RELEASE_KEY_BASE64 }}
```
Comment (mafredri) [inline .github/workflows/release.yaml:217]: Just curious, why base64 vs armored?

File: scripts/release/publish.sh [hunk 2]
```diff
@@ -114,9 +114,50 @@ done
...
+	# Sign the checksums file. This generates a file in the same directory and
+	# with the same name as the checksums file but ending in ".asc".
+	#
+	# We pipe `true` into `gpg` so that it never tries to be interactive (i.e.
+	# ask for a passphrase). The key we import above is not password protected.
+	true | gpg --detach-sign --armor "${temp_dir}/${checksum_file}" 1>&2
+
+	rm -rf "$gnupg_home_temp"
+	if [[ "$old_gnupg_home" == "" ]]; then
+		unset GNUPGHOME
```
Comment (mafredri) [inline scripts/release/publish.sh:142]: This could probably be outside the conditional, since we rm the folder anyway.

**Review by mafredri — APPROVED** [review]
LGTM!

**Comment by mafredri** [comment]
Just realized I didn't see if we upload the file as well, or is it automatic as part of all-files-in-a-dir?

**Comment by deansheather** [comment]
Yep, the publish script automatically uploads all files in `"$temp_dir"/` when calling the `gh release` command

File: .github/workflows/release.yaml (hunk 1 above)
Comment (deansheather) [reply .github/workflows/release.yaml:217]: It's armored *and* base64'd. Base64'd mostly so it's on a single line. It's not necessary, but avoids any weird escaping issues if they ever crop up in the future.

**Review by kylecarbs — APPROVED** [review]

**Pull request merged**